### PR TITLE
Add new properties to the button component developer specified class name and a variant

### PR DESCRIPTION
**Pull request by byohannes**
fix for this issue: https://github.com/Capgemini/dcx-react-library/issues/466

### CODE REVIEW

File: src/button/__tests__/Button.test.tsx [hunk 1]
```diff
@@ -247,4 +247,32 @@ describe('Button', () => {
...
   });
+
+  it('should render the default className dcx-button', () => {
+    const handleClick = jest.fn();
+    render(<Button onClick={handleClick} label="Register" />);
+    const button: any = screen.getByRole('button');
+    expect(button.getAttribute('class')).toBe('dcx-button');
+  });
+
+  it('should render the primary variant to the className', () => {
```
Comment (daniele-zurico) [inline src/button/__tests__/Button.test.tsx:258]: can we try to cover this scenario as well?
```
the developer specifies his own class but not a variant:
<button class="dcx-button daniele">
```

Comment (byohannes) [reply src/button/__tests__/Button.test.tsx:265]: I have covered the scenario as stated @daniele-zurico

**Review by daniele-zurico — DISMISSED** [review]

File: src/button/Button.tsx [hunk 2]
```diff
@@ -68,6 +68,14 @@ type ButtonProps = React.HTMLAttributes<HTMLButtonElement> & {
    * allows to use a value in the button
    */
   value?: string | number;
+  /**
+   * allow the user to define the buttons class name
+   */
+  className?: string;
+    /**
```
Comment (Ibabalola) [inline src/button/Button.tsx:75]:
```suggestion
 /**
```

File: src/button/Button.tsx [hunk 3]
```diff
@@ -127,6 +137,16 @@ export const Button = ({
     if (customPostfixImg) postfix = <>{customPostfixImg}</>;
   }
 
+  const btnClassName = classNames([
+    'dcx-button',
+    className,
+    {
```
Comment (Ibabalola) [inline src/button/Button.tsx:143]: hmmm wondering if this will work also
```jsx
{
  'dcx-button--${variant}':  variant !== undefined,
}
```

**Review by daniele-zurico — DISMISSED** [review]

**Review by daniele-zurico — DISMISSED** [review]

**Review by Ibabalola — APPROVED** [review]

File: src/button/Button.tsx (hunk 3 above)
Comment (byohannes) [reply src/button/Button.tsx:143]: It does work and is more readable and more straightforward as well

Comment (byohannes) [reply src/button/Button.tsx:143]: I have updated it as per your comment.  

[`dcx-button--${variant}`]:  variant !== undefined,

**Review by daniele-zurico — APPROVED** [review]

**Pull request merged**